diff --git a/zutil.c b/zutil.c
--- a/zutil.c
+++ b/zutil.c
@@ -299,7 +299,7 @@ extern void free(voidpf ptr);
 
 voidpf ZLIB_INTERNAL zcalloc(voidpf opaque, unsigned items, unsigned size) {
     (void)opaque;
-    return sizeof(uInt) > 2 ? (voidpf)malloc(items * size) :
+    return sizeof(uInt) > 2 ? (voidpf)malloc((size_t) items * size) :
                               (voidpf)calloc(items, size);
 }
 
